File: Dhiva/moviecatalogue-sub2/app/src/test/java/id/dhiva/moviecatalogue/ui/movie/MovieViewModelTest.java
package id.dhiva.moviecatalogue.ui.movie;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TestRule;

import java.util.ArrayList;

import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
import androidx.lifecycle.Observer;
import id.dhiva.moviecatalogue.R;
import id.dhiva.moviecatalogue.data.DataRepository;
import id.dhiva.moviecatalogue.model.ModelItem;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class MovieViewModelTest {
    private MovieViewModel movieViewModel;
    private DataRepository dataRepository = mock(DataRepository.class);

    @Before
    public void setUp(){
        movieViewModel = new MovieViewModel(dataRepository);
    }

    @Rule
    public TestRule rule = new InstantTaskExecutorRule();

    @Test
    public void getMovies() {
        ArrayList<ModelItem> list = new ArrayList<>();
        ModelItem modelItem = new ModelItem();
        modelItem.setTitle("A Star");
        modelItem.setDate("25 Oktobr 2018");
        modelItem.setDesc("deskripsi");
        modelItem.setGenre("romance");
        modelItem.setRating("8");
        modelItem.setPoster(R.drawable.poster_a_star);
        list.add(modelItem);

        when(dataRepository.getAllMovies()).thenReturn(list);

        Observer<ArrayList<ModelItem>> observer = mock(Observer.class);
        movieViewModel.getAllMovieList().observeForever(observer);
        verify(observer).onChanged(list);
    }
}

File: Submission2/app/src/test/java/cikal/dicoding/submission2/ui/movie/MovieViewModelTest.kt
package cikal.dicoding.submission2.ui.movie

import androidx.arch.core.executor.testing.InstantTaskExecutorRule
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer
import cikal.dicoding.submission2.data.source.CatalogueRepository
import cikal.dicoding.submission2.data.source.local.entity.MovieEntity
import cikal.dicoding.submission2.utils.DataDummy
import org.junit.Assert.*
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.Mockito
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class MovieViewModelTest {

    private lateinit var viewModel: MovieViewModel

    @get:Rule
    var instantTaskExecutorRule = InstantTaskExecutorRule()

    @Mock
    private lateinit var catalogueRepository: CatalogueRepository

    @Mock
    private lateinit var observer: Observer<List<MovieEntity>>

    @Before
    fun setUp() {
        viewModel = MovieViewModel(catalogueRepository)
    }

    @Test
    fun getListMovies() {
        val dummyMovies = DataDummy.generateDummyMovies()
        val movies = MutableLiveData<List<MovieEntity>>()
        movies.value = dummyMovies

        Mockito.`when`(catalogueRepository.getAllMovies()).thenReturn(movies)
        val movieEntities = viewModel.getMovies().value
        Mockito.verify<CatalogueRepository>(catalogueRepository).getAllMovies()
        assertNotNull(movieEntities)
        assertEquals(10, movieEntities?.size)

        viewModel.getMovies().observeForever(observer)
        Mockito.verify(observer).onChanged(dummyMovies)
    }
}